### ISSUE TITLE
Code for Figure 2

### ISSUE THREAD

**Issue by PascalIversen** [zhouyuru1205/PharmaFormer#4]
Hey again,
Could you provide the code for Figure 2? The difference between the baseline performances and those of PharmaFormer appears to be extremely high. Such an extreme difference is pretty unheard of to me in the drug response prediction field. 

Could it be that the Pearson correlation for the PharmaFormer was calculated globally instead of per-drug? 

A global calculation is for example done in the code here:
`        pearson_corr = np.corrcoef(all_outputs, all_targets)[0, 1]
`

<img width="2000" height="2037" alt="Image" src="https://github.com/user-attachments/assets/11da4c11-ec00-4f2b-9e88-dd735fdc82d4" />

Thanks!! :)

**Comment by wconnell** [comment]
> We have also reached out to you via email with a detailed response addressing all of your questions.

What's the response? I have the same questions. Thanks!

**Comment by PascalIversen** [comment]
@wconnell My concerns were not addressed. The authors agreed that there were evaluation errors, but they did not correct the scientific record. The paper, as it stands, is highly misleading. To be honest, I am disappointed that the authors did not ask the journal for a retraction or correction. 

 We also implemented this model in our evaluation pipeline, and it is one of the worst models. On a properly normalized Pearson correlation metric, it is worse than predicting the mean. For the results, see here: https://github.com/daisybio/drevalpy

**Comment by wconnell** [comment]
@PascalIversen Thanks for following up - and great work with @JudithBernett on DrEval! The bio/ml community desperately needs more rigorous scientific benchmarking. 

Frankly, the journal needs to be held accountable as well. There was obviously no diligence in review over such a primary concept of correct stratification for evaluation of the method.

**Comment by zhouyuru1205** [comment]
Hi Pascal,

Thank you again for raising this question and for benchmarking PharmaFormer in DREvalPy.

We checked the DREvalPy PharmaFormer wrapper and identified a prediction-time preprocessing error that substantially affects the evaluation.

In `PharmaFormerModel.predict()`, the preprocessing loop iterates over response-level `cell_line_ids`, while the underlying expression vector is shared at the cell-line level. When a cell line occurs multiple times in one prediction request, its expression vector is repeatedly transformed by `StandardScaler -> MinMaxScaler` and overwritten after each transformation.

We have reported this wrapper error, together with a minimal correction, here:

**(https://github.com/daisybio/drevalpy/issues/475)**

We re-ran the CTRPv2 10-fold leave-cell-out evaluation with just one code correction: replacing `for cell_line_id in cell_line_ids:` with `for cell_line_id in set(cell_line_ids):`. **The dataset, data splits, input features, hyperparameters, and evaluation metric remained unchanged.**

<img width="4290" height="2600" alt="Image" src="https://github.com/user-attachments/assets/c3f6a5c3-405c-4f3a-8d28-805f0d342fc5" />

Using the same 10 data splits and hyperparameters, but independently trained models, our local mean normalized Pearson correlation changed from:

- **0.0065 with the original DREvalPy wrapper**
- **0.3589 with the corrected DREvalPy wrapper**

To isolate the error independently of retraining, we also tested the same checkpoint on the same 48 cell lines and the same drug, changing only the number of occurrences of each cell line in the prediction request. With the original wrapper, the variation in predictions across cell lines collapsed as the occurrence count increased. With the corrected wrapper, predictions remained unchanged. This confirms that the original wrapper makes a prediction depend on the other records included in the request.

Could you please check whether this error affected the results you reported and re-run the evaluation with the corrected wrapper?

Finally, the DREvalPy benchmark and our original experiments differ in the dataset (CTRPv2 versus GDSC), response variable (LN_IC50 versus AUC), metric (normalized Pearson versus mean per-drug Pearson), gene-expression feature set, and cross-validation protocol. We understand the need for standardized comparison conditions in a benchmark. However, these choices can affect model performance, and we have not isolated their individual effects in the comparison above: we corrected only the wrapper's repeated preprocessing and retained the benchmark inputs and settings. The benchmark results therefore characterize PharmaFormer under that particular protocol; they should not be treated as a direct reproduction of our original experiments or as a complete characterization of its performance across evaluation settings.

**Comment by PascalIversen** [comment]
Thanks, good catch, that explains some of the gap. We will fix it and remove it from the leaderboard, and maybe rerun at some point. (The model is very large, so I am not sure about this)
Note that the performance you posted is still lower than that of a Naive model, which just exploits dataset biases.

Regarding the differences between the benchmarking setup: We have support for GDSC and also the AUC measure. What differences in the cross-validation and feature set do you see?

I think a good way to resolve this would be to run a version of the PharmaFormer you think is faithful using AUC on GDSC in DrEval, and then see if it outperforms off-the-shelf ML models/simple feed-forward neural networks on mean per-drug Pearson, right? 

In the meantime, could you please issue a correction for your paper? Figure 2 is still wrong

**Comment by zhouyuru1205** [comment]
Hi Pascal,
Thank you for confirming the preprocessing bug and for planning to address it. We also appreciate your benchmarking suggestion.
We are in contact with the journal’s editors.
